pkg/bib/build.go

```diff
@@ -0,0 +1,205 @@
+		return fmt.Errorf("pulling image: %w", err)
+	}
+	imageFullName := imageInspect.RepoTags[0]
+
+	if buildOption.BibContainerImage == "" {
+		label, found := imageInspect.Labels["bootc.diskimage-builder"]
+		if found && label != "" {
+			buildOption.BibContainerImage = label
+		} else {
+			buildOption.BibContainerImage = defaultBibImage
+		}
+	}
+
+	// Let's pull the Bootc Image Builder if necessary
+	_, err = utils.PullAndInspect(ctx, buildOption.BibContainerImage)
+	if err != nil {
+		return fmt.Errorf("pulling bootc image builder image: %w", err)
+	}
+
+	// BIB doesn't work with just the image ID or short name, it requires the image full name
+	bibContainer, err := createBibContainer(ctx, buildOption.BibContainerImage, imageFullName, buildOption)
+	if err != nil {
+		return fmt.Errorf("failed to create image builder container: %w", err)
+	}
+
+	err = containers.Start(ctx, bibContainer.ID, &containers.StartOptions{})
+	if err != nil {
+		return fmt.Errorf("failed to start image builder container: %w", err)
+	}
+
+	// Ensure we've cancelled the container attachment when exiting this function, as
+	// it takes over stdout/stderr handling
+	attachCancelCtx, cancelAttach := context.WithCancel(ctx)
+	defer cancelAttach()
+
+	if !quiet {
+		attachOpts := new(containers.AttachOptions).WithStream(true)
+		if err := containers.Attach(attachCancelCtx, bibContainer.ID, os.Stdin, os.Stdout, os.Stderr, nil, attachOpts); err != nil {
+			return fmt.Errorf("attaching image builder container: %w", err)
+		}
+	}
+
+	exitCode, err := containers.Wait(ctx, bibContainer.ID, nil)
+	if err != nil {
+		return fmt.Errorf("failed to wait for image builder container: %w", err)
+	}
+
+	if exitCode != 0 {
+		return fmt.Errorf("failed to run image builder")
+	}
+
+	return nil
+}
+
+func createBibContainer(ctx context.Context, bibContainerImage, imageFullName string, buildOption BuildOption) (types.ContainerCreateResponse, error) {
+	privileged := true
+	autoRemove := true
+	labelNested := true
+	terminal := true // Allocate pty so we can show progress bars, spinners etc.
+
+	bibArgs := bibArguments(imageFullName, buildOption)
+
+	s := &specgen.SpecGenerator{
+		ContainerBasicConfig: specgen.ContainerBasicConfig{
+			Remove:       &autoRemove,
+			Annotations:  map[string]string{"io.podman.annotations.label": "type:unconfined_t"},
+			Terminal:     &terminal,
+			Command:      bibArgs,
+			SdNotifyMode: "container", // required otherwise crun will fail to open the sd-bus
+		},
+		ContainerStorageConfig: specgen.ContainerStorageConfig{
+			Image: bibContainerImage,
+			Mounts: []specs.Mount{
+				{
+					Source:      buildOption.Config,
+					Destination: "/config.toml",
+					Type:        "bind",
+				},
+				{
+					Source:      buildOption.Output,
+					Destination: "/output",
+					Type:        "bind",
+					Options:     []string{"nosuid", "nodev"},
+				},
+				{
+					Source:      "/var/lib/containers/storage",
+					Destination: "/var/lib/containers/storage",
```

Comment:
I'm assuming that this command is always running in sudo / root and we shouldn't use `~/.local/share/containers/storage/` right?

This would cause an issue if a user did `podman pull` not `sudo podman pull` before running podman-bootc.

Unsure to be honest how this should be worked around / I guess we assume that podman-bootc is always running privileged?

Reply:
> I guess we assume that podman-bootc is always running privileged?

No; we assume the container is running privileged inside podman machine.